bin/cylc-cat-log

```diff
@@ -31,131 +32,200 @@ from cylc.suite_host import is_remote_host
 from cylc.suite_output import suite_output
 from cylc.cfgspec.globalcfg import GLOBAL_CFG
-import cylc.flags
 from cylc.task_id import TaskID
 
-parser = cop( """cylc [info] cat-log|log [OPTIONS] ARGS
-Print various log files for suites and tasks that are currently running,
-or have previously finished.""",
-    argdoc=[ ("REG", "Suite name"),
-    ("[TASK-ID]", """Print the stdout or stderr log
-of the identified task""" )])
-
-parser.add_option( "-l", "--location",
-        help="Just print the location of the requested log file.",
-        action="store_true", default=False, dest="print_location" )
-
-parser.add_option( "-r", "--rotation",
-        help="Rotation number (to view older, rotated suite logs)",
-        metavar="INT", action="store", default="0", dest="rotation" )
-
-parser.add_option( "-o", "--stdout",
-        help="Print suite or task stdout logs "
-        "(for suites, the default is to print the event log; "
-        " for tasks, the default is to print the job script).",
-        action="store_true", default=False, dest="print_stdout" )
-
-parser.add_option( "-e", "--stderr",
-        help="Print suite or task stderr logs (see --stdout for defaults).",
-        action="store_true", default=False, dest="print_stderr" )
-
-parser.add_option( "-t", "--try-number",
-        help="Task try number (default 1).",
-        metavar='INT', action="store", default="1", dest="tryext" )
-
-( options, args ) = parser.parse_args()
-
-suite, suiterc = prep_file( args[0], options ).execute()
+parser = cop(
+    """cylc [info] cat-log|log [OPTIONS] ARGS
+Print the location or content of any suite or task log file, or a listing of a
+task log directory on the suite or task host.  By default the suite event log
+or task job script is printed.""",
+    argdoc=[("REG", "Suite name"), ("[TASK-ID]", """Task ID""")])
+
+parser.add_option(
+    "-l", "--location",
+    help="Print the location of the file, and set exit status according"
+    "to whether or not the file exists (0 means it exists).",
+    action="store_true", default=False, dest="location")
+
+parser.add_option(
+    "-o", "--stdout",
+    help="Suite or task stdout log.",
+    action="store_true", default=False, dest="stdout")
+
+parser.add_option(
+    "-e", "--stderr",
+    help="Suite or task stderr log.",
+    action="store_true", default=False, dest="stderr")
+
+parser.add_option(
+    "-r", "--rotation",
+    help="Suite logs log rotation number (default 0)",
+    metavar="INT", action="store", default="0", dest="rotation")
+
+parser.add_option(
+    "-a", "--activity",
+    help="Task job activity log.",
+    action="store_true", default=False, dest="job_activity")
+
+parser.add_option(
+    "-d", "--diff",
+    help="Task job script diff (present after an edit-run).",
+    action="store_true", default=False, dest="job_diff")
+
+parser.add_option(
+    "-u", "--status",
+    help="Task job status file.",
+    action="store_true", default=False, dest="job_status")
```

Comment:
It is possible to use only `options.job_custom`:

```python
# ...
parser.add_option(
    "-d", "--diff",
    help="Task job script diff (present after an edit-run).",
    action="store_const", const="job-edit.diff", dest="job_custom")
parser.add_option(
    "-u", "--status",
    help="Task job status file.",
    action="store_const", const="job.status", dest="job_custom")

# ... and so on...

# ... so later on, the "if" branch can be a bit simpler:

if options.job_custom:
    fpath = os.path.join(log_dir, options.job_custom)
else:
    fpath = os.path.join(log_dir, "job")
```

(However, I guess it is not as straightfoward because the logic is overloading the `-e` and `-o` option for both suite and task.)